src/execution/lazy/statements.rs

```diff
@@ -35,10 +36,18 @@ impl LazyStatement {
         debug!("eval {}", self);
         trace!("{{");
         let result = match self {
-            Self::AddGraphNodeAttribute(stmt) => stmt.evaluate(exec),
-            Self::CreateEdge(stmt) => stmt.evaluate(exec),
-            Self::AddEdgeAttribute(stmt) => stmt.evaluate(exec),
-            Self::Print(stmt) => stmt.evaluate(exec),
+            Self::AddGraphNodeAttribute(stmt) => stmt
+                .evaluate(exec)
+                .with_context(|| stmt.debug_info.clone().into()),
```

Comment:
During the second phase, where the lazy values are evaluated, the stored context is attached to any errors that are thrown. Here when a statement is executed.